pkg/parser/parser.go

```diff
@@ -0,0 +1,121 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type parser struct {
+	tokens chan token.Token
+
+	current token.Token
+	peek    token.Token
+
+	errors []error
+}
+
+func New(t chan token.Token) *parser {
+	return &parser{
+		tokens: t,
+		errors: []error{},
+	}
+}
+
+func (p *parser) Errors() []error {
+	return p.errors
+}
+
+func (p *parser) nextToken() {
+	p.current = p.peek
+	p.peek = <-p.tokens
+}
+
+func (p *parser) ParseProgram() *ast.Program {
+	program := &ast.Program{
+		Statements: []ast.Statement{},
+	}
+
+	for !p.currentIs(token.EOF) {
+		stmt := p.ParseStatement()
+		if stmt != nil {
+			program.Statements = append(program.Statements, stmt)
+		}
+
+		p.nextToken()
+	}
+
+	return program
+}
+
+func (p *parser) ParseStatement() ast.Statement {
```

Comment:
Do you need this to be exported? I don't think its needed outside this package...

Reply:
Same applies to all other `parserXxxStatement`

Reply:
Yeah, they don't need to be exported.